kat-imp.md

```diff
@@ -71,10 +71,9 @@ module IMP-BIMC
          ...
          </s>
 
-    syntax StatePred ::= "div-zero-error?"
- // --------------------------------------
-    rule <s> div-zero-error? [ <imp> <k> div-zero-error ... </k> ... </imp> ] => #true  ... </s>
-    rule <s> div-zero-error? [ <imp> <k> NEXT:K         ... </k> ... </imp> ] => #false ... </s> requires NEXT =/=K div-zero-error
+    syntax Pred ::= "div-zero-error?"
+ // ---------------------------------
+    rule <s> div-zero-error? => can? (^ divzero) ... </s>
```

Comment:
So the idea is now this rule is general and user just has to specify `divzero` for this to work? I'm confused why this is better, since before the user would just have to specify `div-zero-error?` anyway (i.e. the user has to write one rule regardless) and now this rule is much less efficient

Reply:
Hmm, I guess you could instead write `bimc N bexp? (not can? ^ divzero)` just as well.

It is a little less efficient, but one positive aspect is that we're using whether a rule can apply as the test instead of `==K` or `=/=K`. This is good because the matching engine is applied in a standard way to check "can this rule apply", whereas `==K` and `=/=K` may not be behaving how I expected. For example, in one case I was able to get `div-zero-error ==K 7 /Int X` (for `X` symbolic) to be satisfiable according to whatever implements `==K`.

The fact that this is now "language independent" (in the sense that you could just supply whichever rule is the `divzero` rule) hadn't occurred to me.

If you like, I can remove this rule altogether, and replace the queries which use `not div-zero-error?` with `not can? ^ div-zero`. I kinda like the idea.

Reply:
Ah ok, avoiding potential issues with `==K` and `=/=K` makes sense, though I'm not sure why `div-zero-error ==K 7 /Int X` shouldn't be satisfiable? Either way, letting the matching engine control as much as possible is a good thing.

As far as whether or not you'd like to remove the rule altogether, I don't really have much of a preference, either is fine with me.

Thanks for clearing that up, looks good to merge now.